### PR TITLE
Detail widget

### PR DESCRIPTION
**Pull request by wongpratan**

### CODE REVIEW

**Review by nh758 — APPROVED** [review]
I didn't read everything, but I think its ok

File: AppBuilder/platform/views/ABViewDetail.js [hunk 1]
```diff
@@ -19,262 +20,44 @@ module.exports = class ABViewDetail extends ABViewDetailCore {
...
     *
     * @return {obj } UI component
     */
-   component(App, idPrefix) {
+   component(v1App, idPrefix) {
+      let component = new ABViewDetailComponent(this);
+
+      // if this is our v1Interface
+      if (v1App) {
+         var newComponent = component;
```
Comment (guyyoo) [inline AppBuilder/platform/views/ABViewDetail.js:45]: var -> const/let

File: AppBuilder/platform/views/ABViewDetailCheckbox.js [hunk 2]
```diff
@@ -22,64 +14,58 @@ module.exports = class ABViewDetailCheckbox extends ABViewDetailCheckboxCore {
...
-         logic: _logic,
-      };
-   }
+   //    var checkboxElem = this.component(App).ui;
+   //    checkboxElem.id = ids.component;
 
-   //
-   // Property Editor
-   //
+   //    var _ui = {
```
Comment (guyyoo) [inline AppBuilder/platform/views/ABViewDetailCheckbox.js:26]: var -> const/let

File: AppBuilder/platform/views/ABViewDetailCheckbox.js [hunk 3]
```diff
@@ -22,64 +14,58 @@ module.exports = class ABViewDetailCheckbox extends ABViewDetailCheckboxCore {
...
     *
     * @return {obj} UI component
     */
-   component(App, idPrefix) {
+   component(v1App) {
+      let component = new ABViewDetailCheckboxComponent(this);
+
+      // if this is our v1Interface
+      if (v1App) {
+         var newComponent = component;
```
Comment (guyyoo) [inline AppBuilder/platform/views/ABViewDetailCheckbox.js:53]: var -> const/let

**Review by guyyoo — APPROVED** [review]
It's good to go.

**Pull request merged**